hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsResourceServerConfiguration.kt

```diff
@@ -0,0 +1,80 @@
+package uk.gov.justice.hmpps.kotlin.auth
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.REACTIVE
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.SERVLET
+import org.springframework.boot.autoconfigure.web.servlet.ConditionalOnMissingFilterBean
+import org.springframework.cache.CacheManager
+import org.springframework.cache.annotation.EnableCaching
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity
+import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.invoke
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.config.web.server.ServerHttpSecurity
+import org.springframework.security.config.web.server.invoke
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
+import org.springframework.security.web.SecurityFilterChain
+import org.springframework.security.web.server.SecurityWebFilterChain
+
+@Configuration
+@ConditionalOnWebApplication(type = SERVLET)
+@EnableWebSecurity
+@EnableMethodSecurity(prePostEnabled = true, proxyTargetClass = true)
+@EnableCaching
+class HmppsResourceServerConfiguration {
+  @ConditionalOnMissingFilterBean
+  @Bean
+  fun hmppsSecurityFilterChain(http: HttpSecurity): SecurityFilterChain =
+    http {
+      sessionManagement { SessionCreationPolicy.STATELESS }
+      headers { frameOptions { sameOrigin = true } }
+      csrf { disable() }
+      authorizeHttpRequests {
+        unauthorizedRequestPaths().forEach { authorize(it, permitAll) }
+        authorize(anyRequest, authenticated)
+      }
+      oauth2ResourceServer {
+        jwt { jwtAuthenticationConverter = AuthAwareTokenConverter() }
+      }
+    }
+      .let { http.build() }
+
+  @ConditionalOnMissingBean
+  @Bean
+  fun locallyCachedJwtDecoder(
```

Comment:
shouldn't this be outside the class in its own one - looks like it will only be wired up in a servlet environment

Reply:
Couldn't find a cache available on the [reactive version](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/jwt/NimbusReactiveJwtDecoder.JwkSetUriReactiveJwtDecoderBuilder.html) of the JWT decoder builder